### PR TITLE
Allow numpy arrays to be passed by value into tasks (and inlined in the task spec).

### PR DESCRIPTION
**Pull request by robertnishihara**
This is related to #1813.

For some reason, this PR doesn't seem to work. E.g., If I do

```python
import ray
import numpy as np

ray.local_scheduler.check_simple_value(np.ones(3))
```

The last line returns `False`.

### CODE REVIEW

**Comment by robertnishihara** [comment]
retest this please

File: src/common/lib/python/common_extension.cc [hunk 1]
```diff
@@ -831,13 +842,26 @@ int is_simple_value(PyObject *value, int *num_elements_contained) {
...
+        return false;
       }
     }
     return (*num_elements_contained <
             RayConfig::instance().num_elements_limit());
   }
-  return 0;
+  if (PyArray_CheckExact(value)) {
+    PyArrayObject *array = reinterpret_cast<PyArrayObject *>(value);
+    if (PyArray_NBYTES(array) > RayConfig::instance().num_elements_limit()) {
```
Comment (pcmoritz) [inline src/common/lib/python/common_extension.cc]: This looks a little confusing, why isn't this subsumed by
```
+    *num_elements_contained += PyArray_NBYTES(array);
+    return (*num_elements_contained <
+            RayConfig::instance().num_elements_limit());
```
below?

Comment (robertnishihara) [reply src/common/lib/python/common_extension.cc]: Fixed, thx.

**Comment by pcmoritz** [comment]
retest this please

**Comment by robertnishihara** [comment]
retest this please

**Comment by robertnishihara** [comment]
For the differences between `import_array` and `import_array1` see https://github.com/numpy/numpy/blob/4cc8cb27e792098875edea9fa000528ca1b34f02/numpy/core/code_generators/generate_numpy_api.py#L130-L134.

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**